Actions/CheckForUpdates/CheckForUpdates.ps1

```diff
@@ -40,21 +40,24 @@ if ($update -eq 'Y') {
     }
 }
 
-$readToken = $token
 if ($token) {
     # token comes from a secret, base 64 encoded
     $token = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($token))
-
-    # Get token with read permissions for this and the template repository - if private and in the same organization
-    $repositories = @($ENV:GITHUB_REPOSITORY)
-    if ($templateUrl -like "https://github.com/$($ENV:GITHUB_REPOSITORY_OWNER)/*") {
-        $repositories += $templateUrl.Split('@')[0]
-    }
-    $readToken = GetAccessToken -token $token -permissions @{"actions"="read";"contents"="read";"metadata"="read"} -repositories $repositories
 }
 
 # Use Authenticated API request if possible to avoid the 60 API calls per hour limit
-$headers = GetHeaders -token $readToken
+$headers = GetHeaders -token $ENV:GITHUB_TOKEN
+$templateRepositoryUrl = $templateUrl.Split('@')[0]
+$response = Invoke-WebRequest -Headers $headers -Method Head -Uri $templateRepositoryUrl -ErrorAction SilentlyContinue
+if (-not $response -or $response.StatusCode -ne 200) {
+    # GITHUB_TOKEN doesn't have access to template repository, must be is private/internal
+    # Get token with read permissions for the template repository
+    $templateRepository = $templateRepositoryUrl.Split('/')[-2..-1] -join '/'
+    $templateReadToken = GetAccessToken -token $token -permissions @{"actions"="read";"contents"="read";"metadata"="read"} -repository $templateRepository -repositories @($templateRepositoryUrl)
```

Comment:
Isn't the `-repository` supposed to be the current repo? How does this work?

Reply:
You are right - that needs to be the repository, where the github app is installed

Reply:
Good catch, I think this also explains why GitHub apps didn't work for dependency resolution - will check this

Reply:
Did a number of tests and it actually needs to be the template repository in -repository.

When getting an access token, repositories specified in -repositories needs to be in the same org as the repository specified by -repository. The repository is the org/repo from which to get the installation, that access token will also have access to all repositories, which needs to be in the same org as repository (and they all need to have the app installed)

So in order to support the template in a different org - it needs to be in repository.

Also, I have tested that I can have dependencies downloaded using a GitHub app from different repositories in the same org and I am testing now that it also works from a different org - again, the repo is specified in -repository for this to work.

Reply:
This is downloading dependencies from BusinessCentralDemos/DepApp2, when the dependencies are in BusinessCentralDemos/DepApp1:
![image](https://github.com/user-attachments/assets/0f426ab3-4787-4add-99b8-5e93192b4f1f)

Reply:
This is downloading dependencies from BusinessCentralDemos/DepApp2 when the dependencies are in BusinessCentralApps/DepApp1 (note the different org):
![image](https://github.com/user-attachments/assets/c0aa415a-41a9-46c0-9d27-f21415143a5d)

Reply:
But, I did find a problem with private repositories as dependencies...

Reply:
Will fix this as part of this PR - https://github.com/microsoft/AL-Go/discussions/1616

That will need to be later...

Reply:
got it - this fixes it https://github.com/microsoft/AL-Go/pull/1663/commits/780135f3ddaa35033f0906d3bd749b099ada9c75